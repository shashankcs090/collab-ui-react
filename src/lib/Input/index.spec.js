import React from 'react';
import { shallow, mount } from 'enzyme';
import {
  Input,
  InputError,
  InputHelper,
  Label,
 } from '@collab-ui/react';


describe('tests for <Input />', () => {
  it('should match text SnapShot', () => {
    const container = shallow(
      <Input htmlId="1" name="test" label="test" type="text" />
    );

    expect(container).toMatchSnapshot();
  });

  it('should match number SnapShot', () => {
    const container = shallow(
      <Input htmlId="1" name="test" label="test" type="number" />
    );

    expect(container).toMatchSnapshot();
  });

  it('should match password SnapShot', () => {
    const container = shallow(
      <Input htmlId="1" name="test" label="test" type="password" />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render one Input', () => {
    const container = shallow(<Input htmlId="1" name="test" label="test" />);

    expect(container.find('input').length).toEqual(1);
    expect(container.children().length).toEqual(2);
  });

  it('should render one Input and apply dirty class', () => {
    const container = shallow(<Input htmlId="1" name="test" label="test" value='test' />);

    expect(container.find('.dirty').length).toEqual(1);
  });

  it('should render Label', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" />
    );

    expect(
      container.contains(
        <Label className="cui-label" htmlFor="test123" label="test" />
      )
    ).toEqual(true);
  });

  it('should pass class based on inputSize prop', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" inputSize="medium-12" />
    );

    expect(container.find('div').hasClass('medium-12')).toEqual(true);
    expect(container.find('div').hasClass('columns')).toEqual(true);
  });

  it('should pass placeholder attribute', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" placeholder="test" />
    );

    expect(container.find('input').props().placeholder).toEqual('test');
  });

  it('should pass disabled attribute', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" disabled />
    );

    expect(container.find('input').props().disabled).toEqual(true);
  });

  it('should pass readOnly attribute', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" readOnly />
    );

    expect(container.find('input').props().readOnly).toEqual(true);
  });

  it('should pass value attribute', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" value="testing" />
    );

    expect(container.find('input').props().value).toEqual('testing');
  });

  it('should pass class based on nesting', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" nestedLevel={1} />
    );

    expect(container.hasClass('cui-input--nested-1')).toEqual(true);
  });

  it('should render Secondary Container and Label', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" secondaryLabel="test" />
    );

    expect(
      container.contains(
        <Label
          className="cui-label__secondary-label"
          htmlFor="test123"
          label="test"
        />
      )
    ).toEqual(true);

    expect(container.find('div.cui-label__secondary-label-container')).toHaveLength(1);
  });

  it('should render Helper Text', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test" inputHelpText="test" />
    );

    expect(container.contains(<InputHelper message="test" />)).toEqual(true);
  });

  it('should determine correct error class(warning)', () => {
    const container = shallow(
      <Input
        htmlId="test123"
        name="test"
        label="test"
        errorArr={[{ error: 'test', type: 'warning' }]}
      />
    );

    expect(container.find('.cui-input-group').hasClass('warning')).toEqual(true);
  });

  it('should determine correct error class(error)', () => {
    const container = shallow(
      <Input
        htmlId="test123"
        name="test"
        label="test"
        errorArr={[{ error: 'test', type: 'error' }]}
      />
    );

    expect(container.find('.cui-input-group').hasClass('error')).toEqual(true);
  });

  it('should determine correct error class if passed error and warning(error)', () => {
    const container = shallow(
      <Input
        htmlId="test123"
        name="test"
        label="test"
        errorArr={[
          { error: 'error1', type: 'error' },
          { error: 'error2', type: 'warning' },
        ]}
      />
    );

    expect(container.find('.cui-input-group').hasClass('error')).toEqual(true);
  });

  it('should render Error Component', () => {
    const container = shallow(
      <Input
        htmlId="test123"
        name="test"
        label="test"
        errorArr={[{ error: 'test', type: 'error' }]}
      />
    );

    expect(container.contains(<InputError error={'test'} />)).toEqual(
      true
    );
  });

  it('should support input ref', () => {
    class Container extends React.Component {
      render() {
        return (
          <Input
            htmlId="test123"
            name="test"
            label="test"
            inputRef={ref => (this.input = ref)}
          />
        );
      }
    }
    const instance = mount(<Container />).instance();

    expect(instance.input.tagName).toEqual('INPUT');
  });

  it('should render children', () => {
    const container = shallow(
      <Input htmlId="test123" name="test" label="test">
        <div className="testingforhoc" />
      </Input>
    );

    expect(container.find('.testingforhoc').length).toEqual(1);
  });

  it('should handle onChange event', () => {
    let count = 0;
    const countUp = () => count++;
    const container = mount(
      <Input htmlId="test" name="test" label="test" onChange={countUp} />
    );

    container.find('input').simulate('change', {target: { value: 'test' }});
    expect(count).toEqual(1);
  });

  it('should handle mouse down event', () => {
    let count = 0;
    const container = shallow(
      <Input htmlId="test" name="test" label="test" onMouseDown={() => count++} />
    );

    container.find('input').simulate('mousedown');
    expect(count).toEqual(1);
  });

  it('should handle onFocus event', () => {
    let count = 0;
    const container = shallow(
      <Input htmlId="test" name="test" label="test" onFocus={() => count++} />
    );

    container.find('input').simulate('focus');
    expect(count).toEqual(1);
  });

  it('should handle onClick event', () => {
    const onClickFn = jest.fn();
    const container = shallow(
        <Input htmlId="test" name="test" label="test" onClick={onClickFn} />
    );
    container.find('input').simulate('click');
    expect(onClickFn).toHaveBeenCalled();
  });
});
