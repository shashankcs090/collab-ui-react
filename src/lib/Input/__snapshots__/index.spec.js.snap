// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <Input /> should match clear SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className=""
    clear={true}
    clearAriaLabel={null}
    defaultValue=""
    disabled={false}
    errorArr={Array []}
    htmlId="1"
    inputClassName=""
    inputHelpText=""
    inputRef={null}
    inputSize=""
    label="test"
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    theme=""
    type="text"
    value=""
  >
    
  </Input>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          className="cui-label"
          htmlFor="1"
          label="test"
          theme=""
        />,
        <div
          className="cui-input__icon-container"
        >
          <input
            className="cui-input"
            disabled={false}
            id="1"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder=""
            readOnly={false}
            tabIndex={0}
            type="text"
            value=""
          />
          
          
        </div>,
        "",
        Array [],
      ],
      "className": "cui-input-group ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "cui-label",
          "htmlFor": "1",
          "label": "test",
          "theme": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="cui-input"
              disabled={false}
              id="1"
              name="test"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              placeholder=""
              readOnly={false}
              tabIndex={0}
              type="text"
              value=""
            />,
            "",
            "",
          ],
          "className": "cui-input__icon-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "cui-input",
              "disabled": false,
              "id": "1",
              "name": "test",
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "onKeyDown": [Function],
              "onMouseDown": [Function],
              "placeholder": "",
              "readOnly": false,
              "tabIndex": 0,
              "type": "text",
              "value": "",
            },
            "ref": [Function],
            "rendered": null,
            "type": "input",
          },
          "",
          "",
        ],
        "type": "div",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            className="cui-label"
            htmlFor="1"
            label="test"
            theme=""
          />,
          <div
            className="cui-input__icon-container"
          >
            <input
              className="cui-input"
              disabled={false}
              id="1"
              name="test"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              placeholder=""
              readOnly={false}
              tabIndex={0}
              type="text"
              value=""
            />
            
            
          </div>,
          "",
          Array [],
        ],
        "className": "cui-input-group ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "cui-label",
            "htmlFor": "1",
            "label": "test",
            "theme": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="cui-input"
                disabled={false}
                id="1"
                name="test"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                placeholder=""
                readOnly={false}
                tabIndex={0}
                type="text"
                value=""
              />,
              "",
              "",
            ],
            "className": "cui-input__icon-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "cui-input",
                "disabled": false,
                "id": "1",
                "name": "test",
                "onBlur": [Function],
                "onChange": [Function],
                "onFocus": [Function],
                "onKeyDown": [Function],
                "onMouseDown": [Function],
                "placeholder": "",
                "readOnly": false,
                "tabIndex": 0,
                "type": "text",
                "value": "",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            "",
            "",
          ],
          "type": "div",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`tests for <Input /> should match number SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className=""
    clear={false}
    clearAriaLabel={null}
    defaultValue=""
    disabled={false}
    errorArr={Array []}
    htmlId="1"
    inputClassName=""
    inputHelpText=""
    inputRef={null}
    inputSize=""
    label="test"
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    theme=""
    type="number"
    value=""
  >
    
  </Input>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          className="cui-label"
          htmlFor="1"
          label="test"
          theme=""
        />,
        <input
          className="cui-input"
          disabled={false}
          id="1"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          placeholder=""
          readOnly={false}
          tabIndex={0}
          type="number"
          value=""
        />,
        "",
        Array [],
      ],
      "className": "cui-input-group ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "cui-label",
          "htmlFor": "1",
          "label": "test",
          "theme": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cui-input",
          "disabled": false,
          "id": "1",
          "name": "test",
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "onKeyDown": [Function],
          "onMouseDown": [Function],
          "placeholder": "",
          "readOnly": false,
          "tabIndex": 0,
          "type": "number",
          "value": "",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            className="cui-label"
            htmlFor="1"
            label="test"
            theme=""
          />,
          <input
            className="cui-input"
            disabled={false}
            id="1"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder=""
            readOnly={false}
            tabIndex={0}
            type="number"
            value=""
          />,
          "",
          Array [],
        ],
        "className": "cui-input-group ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "cui-label",
            "htmlFor": "1",
            "label": "test",
            "theme": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cui-input",
            "disabled": false,
            "id": "1",
            "name": "test",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onKeyDown": [Function],
            "onMouseDown": [Function],
            "placeholder": "",
            "readOnly": false,
            "tabIndex": 0,
            "type": "number",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`tests for <Input /> should match password SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className=""
    clear={false}
    clearAriaLabel={null}
    defaultValue=""
    disabled={false}
    errorArr={Array []}
    htmlId="1"
    inputClassName=""
    inputHelpText=""
    inputRef={null}
    inputSize=""
    label="test"
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    theme=""
    type="password"
    value=""
  >
    
  </Input>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          className="cui-label"
          htmlFor="1"
          label="test"
          theme=""
        />,
        <input
          className="cui-input"
          disabled={false}
          id="1"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          placeholder=""
          readOnly={false}
          tabIndex={0}
          type="password"
          value=""
        />,
        "",
        Array [],
      ],
      "className": "cui-input-group ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "cui-label",
          "htmlFor": "1",
          "label": "test",
          "theme": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cui-input",
          "disabled": false,
          "id": "1",
          "name": "test",
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "onKeyDown": [Function],
          "onMouseDown": [Function],
          "placeholder": "",
          "readOnly": false,
          "tabIndex": 0,
          "type": "password",
          "value": "",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            className="cui-label"
            htmlFor="1"
            label="test"
            theme=""
          />,
          <input
            className="cui-input"
            disabled={false}
            id="1"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder=""
            readOnly={false}
            tabIndex={0}
            type="password"
            value=""
          />,
          "",
          Array [],
        ],
        "className": "cui-input-group ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "cui-label",
            "htmlFor": "1",
            "label": "test",
            "theme": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cui-input",
            "disabled": false,
            "id": "1",
            "name": "test",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onKeyDown": [Function],
            "onMouseDown": [Function],
            "placeholder": "",
            "readOnly": false,
            "tabIndex": 0,
            "type": "password",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`tests for <Input /> should match text SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    className=""
    clear={false}
    clearAriaLabel={null}
    defaultValue=""
    disabled={false}
    errorArr={Array []}
    htmlId="1"
    inputClassName=""
    inputHelpText=""
    inputRef={null}
    inputSize=""
    label="test"
    name="test"
    nestedLevel={0}
    onChange={null}
    onDoneEditing={null}
    onFocus={null}
    onKeyDown={null}
    onMouseDown={null}
    placeholder=""
    readOnly={false}
    secondaryLabel=""
    theme=""
    type="text"
    value=""
  >
    
  </Input>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Label
          className="cui-label"
          htmlFor="1"
          label="test"
          theme=""
        />,
        <input
          className="cui-input"
          disabled={false}
          id="1"
          name="test"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          placeholder=""
          readOnly={false}
          tabIndex={0}
          type="text"
          value=""
        />,
        "",
        Array [],
      ],
      "className": "cui-input-group ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "cui-label",
          "htmlFor": "1",
          "label": "test",
          "theme": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cui-input",
          "disabled": false,
          "id": "1",
          "name": "test",
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "onKeyDown": [Function],
          "onMouseDown": [Function],
          "placeholder": "",
          "readOnly": false,
          "tabIndex": 0,
          "type": "text",
          "value": "",
        },
        "ref": [Function],
        "rendered": null,
        "type": "input",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Label
            className="cui-label"
            htmlFor="1"
            label="test"
            theme=""
          />,
          <input
            className="cui-input"
            disabled={false}
            id="1"
            name="test"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            placeholder=""
            readOnly={false}
            tabIndex={0}
            type="text"
            value=""
          />,
          "",
          Array [],
        ],
        "className": "cui-input-group ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "cui-label",
            "htmlFor": "1",
            "label": "test",
            "theme": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cui-input",
            "disabled": false,
            "id": "1",
            "name": "test",
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onKeyDown": [Function],
            "onMouseDown": [Function],
            "placeholder": "",
            "readOnly": false,
            "tabIndex": 0,
            "type": "text",
            "value": "",
          },
          "ref": [Function],
          "rendered": null,
          "type": "input",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
